src/pages/groupComparison/Survival.tsx

```diff
@@ -26,12 +27,51 @@ export default class Survival extends React.Component<ISurvivalProps, {}> {
             this.props.store.uidToGroup
         ],
         invoke: () => {
+            const orderedActiveGroupUidSet = _.reduce(this.props.store._activeGroupsNotOverlapRemoved.result!, (acc, next, index) => {
+                acc[next.uid] = index;
+                return acc;
+            }, {} as { [id: string]: number });
             const partition = this.props.store.patientsVennPartition.result!;
+
+            // ascending sort partition bases on number of groups in each parition.
+            // if they are equal then sort based on the give order of groups
+            partition.sort((a, b) => {
+                let a1Keys = _.reduce(a.key, (acc, next, key) => {
```

Comment:
I think this is a little hard to read - Object.keys(a.key).filter(uid=>a.key[uid]) is one line and clearer

Reply:
also `a1Keys` is not a good variable name - should be `aUids` or something

Reply:
ha.  i wrote the exact opposite.  ok.  not important enough to fight about